src/sentry/apidocs/extensions.py

```diff
@@ -0,0 +1,18 @@
+from drf_spectacular.extensions import OpenApiAuthenticationExtension
+
+
+class TokenAuthExtension(OpenApiAuthenticationExtension):
```

Comment:
this will automatically document the permission scopes that are needed for the endpoint. there is some tweaks to be done, but it will basically work like this

Reply:
may be worth adding some tests for this extension, but i'm 50/50 on it for now.